src/SignalR/common/Shared/ReflectionHelper.cs

```diff
@@ -38,6 +38,11 @@ public static bool IsStreamingType(Type type, bool mustBeDirectType = false)
 #if NETCOREAPP3_0
         public static bool IsIAsyncEnumerable(Type type)
         {
+            if (type.IsGenericType)
```

Comment:
Does the current way not detect `IAsncEnumerable` correctly?

Reply:
Correct. Previously, `ReflectionHelper.IsIAsyncEnumerable(typeof(IAsyncEnumerable<Foo>))` would return false because `typeof(IAsyncEnumerable<Foo>).GetInterfaces()` returns an empty array.

We mostly used to call IsIAsyncEnumerable with the result of object.GetType() prior to this PR to find streaming parameters in server-to-client invocations. In that case, the type would have to be concrete, and the old version of IsIAsyncEnumerable did the job.

This was and is still [called indirectly from HubMethodDescriptor](https://github.com/aspnet/AspNetCore/blob/4ddf91c09261436bcca1c0de2bb56269b6c229d4/src/SignalR/server/Core/src/Internal/HubMethodDescriptor.cs), and there the type could be an interface, so this had to be fixed to make IAsyncEnumerable hub parameters to work.

Reply:
Might be worth added some tests for this